packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,670 @@
+      batchJob.context
+
+    const {
+      select: filterableFields,
+      skip,
+      take,
+      ...listConfig
+    } = prepareListQuery(
+      {
+        context: batchJob.context,
+        type: batchJob.type,
+        dry_run: batchJob.dry_run,
+        limit: limit as number,
+        offset: offset as number,
+        order: order as string,
+        fields: fields as string,
+        expand: expand as string,
+      },
+      {
+        isList: true,
+        defaultRelations: this.relations_,
+      }
+    )
+
+    batchJob.context = {
+      ...(context ?? {}),
+      listConfig,
+      filterableFields,
+      offset: skip,
+    }
+
+    return batchJob
+  }
+
+  async processJob(batchJobId: string): Promise<void> {
+    let offset = 0
+    let advancementCount = 0
+    let productCount = 0
+
+    return await this.atomicPhase_(
+      async (transactionManager) => {
+        let batchJob = await this.batchJobService_
+          .withTransaction(transactionManager)
+          .retrieve(batchJobId)
+
+        const {
+          writeStream,
+          url: key,
+          promise,
+        } = await this.fileService_.getUploadStreamDescriptor({
+          name: "product-export",
+          ext: "csv",
+        })
+
+        const header = await this.buildHeader()
+        writeStream.write(header)
+
+        offset = batchJob.context.offset ?? offset
+        advancementCount = batchJob.result.advancementCount ?? advancementCount
+        const { listConfig, filterableFields } = batchJob.context as Context
+
+        const [productList, count] = await this.productService_
+          .withTransaction(transactionManager)
+          .listAndCount(filterableFields, {
+            ...(listConfig ?? {}),
+            skip: offset,
+            take: this.BATCH_SIZE,
+          })
+
+        productCount = count
+        let products = productList
+
+        while (offset < productCount) {
+          if (!products?.length) {
+            products = await this.productService_
+              .withTransaction(transactionManager)
+              .list(filterableFields, {
+                ...(listConfig ?? {}),
+                skip: offset,
+                take: this.BATCH_SIZE,
+              })
+          }
+
+          products.forEach((product: Product) => {
+            const lines = this.buildProductVariantLines(product)
+            lines.forEach((line) => writeStream.write(line))
+          })
+
+          advancementCount += products.length
+          offset += products.length
+          products.length = 0
+
+          batchJob = await this.batchJobService_.update(batchJobId, {
+            context: {
+              ...batchJob.context,
+              fileKey: key,
+              offset,
+            },
+            result: {
+              count: productCount,
+              advancementCount,
+              progress: advancementCount / productCount,
+              err: undefined,
+            },
```

Comment:
comment: To be modified if other suggestion is incorporated
```suggestion
            context: {
              ...batchJob.context,
              file_key: key,
              offset,
            },
            result: {
              count: productCount,
              advancement_count: advancementCount,
              progress: advancementCount / productCount,
              err: undefined,
            },
```